devito/core/autotuning.py

```diff
@@ -34,14 +40,21 @@ def autotune(operator, arguments, tunable, mode='basic'):
     if mode == 'aggressive':
         blocksizes = more_heuristic_attempts(blocksizes)
 
+    # How many temporaries are allocated on the stack?
+    # Will drop block sizes that might lead to a stack overflow
+    functions = FindSymbols('symbolics').visit(operator.body +
+                                               operator.elemental_functions)
+    stack_shapes = [i.shape for i in functions if i.is_TensorFunction and i._mem_stack]
+    stack_space = sum(reduce(mul, i, 1) for i in stack_shapes)*operator.dtype().itemsize
+
     # Note: there is only a single loop over 'blocksize' because only
     # square blocks are tested
     timings = OrderedDict()
-    for blocksize in blocksizes:
+    for bs in blocksizes:
```

Comment:
Ah, for loop iteration variables, I accepted abbreviations or simple indices such as ``i``